File: Leetcode/Easy/Defanging an IP Address/defangIPaddr.js
/**
 * @param {string} address
 * @return {string}
 */
function defangIPaddr(address) {
  return address.replaceAll(".", "[.]");
}

let testStr = defangIPaddr("1.1.1.1");

console.log(testStr);

let address = "1[.]1[.]1[.]1"

console.log(testStr === address);

File: Leetcode/Easy/Defanging an IP Address/defangIPaddr.c
#include <stdlib.h>
#include <string.h>

char *defangIPaddr(char *address) {
  char *result = malloc(strlen(address) * 3 + 1);
  char *p = result;

  while (*address) {
    if (*address == '.') {
      *p++ = '[';
      *p++ = '.';
      *p++ = ']';
    } else {
      *p++ = *address;
    }
    address++;
  }

  *p = '\0';
  return result;
}
